cmd/bbolt/main_test.go

```diff
@@ -277,38 +277,78 @@ func TestBucketsCommand_Run(t *testing.T) {
 
 // Ensure the "keys" command can print a list of keys for a bucket.
 func TestKeysCommand_Run(t *testing.T) {
-	db := btesting.MustCreateDB(t)
+	t.Run("test keys in string format", func(t *testing.T) {
+		db := btesting.MustCreateDB(t)
+		if err := db.Update(func(tx *bolt.Tx) error {
+			for _, name := range []string{"foo", "bar"} {
+				b, err := tx.CreateBucket([]byte(name))
+				if err != nil {
+					return err
+				}
+				for i := 0; i < 3; i++ {
+					key := fmt.Sprintf("%s-%d", name, i)
+					if err := b.Put([]byte(key), []byte{0}); err != nil {
+						return err
+					}
+				}
+			}
+			return nil
+		}); err != nil {
+			t.Fatal(err)
+		}
+		db.Close()
 
-	if err := db.Update(func(tx *bolt.Tx) error {
-		for _, name := range []string{"foo", "bar"} {
-			b, err := tx.CreateBucket([]byte(name))
+		defer requireDBNoChange(t, dbData(t, db.Path()), db.Path())
+
+		expected := "foo-0\nfoo-1\nfoo-2\n"
+
+		// Run the command.
+		m := NewMain()
+		if err := m.Run("keys", db.Path(), "foo"); err != nil {
+			t.Fatal(err)
+		} else if actual := m.Stdout.String(); actual != expected {
+			t.Fatalf("unexpected stdout:\n\n%s", actual)
+		}
+	})
+
+	t.Run("test keys in int format", func(t *testing.T) {
+		db := btesting.MustCreateDB(t)
+		if err := db.Update(func(tx *bolt.Tx) error {
+			b, err := tx.CreateBucket([]byte("test-bucket"))
 			if err != nil {
 				return err
 			}
-			for i := 0; i < 3; i++ {
-				key := fmt.Sprintf("%s-%d", name, i)
-				if err := b.Put([]byte(key), []byte{0}); err != nil {
+			for _, k := range []int64{10001, 10002} {
```

Comment:
@ahrtr  I added this sub-test 

- insert keys in int64 format
- asserting the keys are retrieved correctly 

running this test fails on 
```
  --- FAIL: TestKeysCommand_Run/test_keys_in_int_format (0.08s)
        btesting.go:47: Opening bbolt DB at: /var/folders/vh/wkbmyt411316pt46h81d6_300000gn/T/TestKeysCommand_Runtest_keys_in_int_format3069027896/001/db
        btesting.go:87: Closing bbolt DB at: /var/folders/vh/wkbmyt411316pt46h81d6_300000gn/T/TestKeysCommand_Runtest_keys_in_int_format3069027896/001/db
        main_test.go:341: unexpected stdout:
            
            a29c01
            a49c01

```